addon/components/o-s-s/side-panel.ts

```diff
@@ -0,0 +1,54 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { run } from '@ember/runloop';
+
+interface OssSidePanelArgs {
+  visible: boolean;
+  backdrop: boolean;
+  size?: 'sm' | 'md';
+  onClose(): void;
+}
+
+export default class OssSidePanel extends Component<OssSidePanelArgs> {
+  @tracked showReturnButton: boolean = false;
+  declare initialTarget: HTMLElement | null;
+
+  get sidePanelSize(): string {
+    if (this.args.size) {
+      return `oss-side-panel--${this.args.size}`;
+    }
+    return 'oss-side-panel--sm';
+  }
+
+  @action
+  onDestroy(): void {
+    run(() => {
+      document.removeEventListener('mousedown', this.trackInitialTarget);
+    });
+  }
+
+  @action
+  initialize(enableListeners: boolean = true): void {
+    if (enableListeners) {
+      run(() => {
+        document.addEventListener('mousedown', this.trackInitialTarget);
+      });
+    }
+  }
+
+  @action
+  trackInitialTarget(event: MouseEvent): void {
```

Comment:
looks like this method is only triggered locally. If true, can you add the `private` keyword ?

Reply:
```
@action
private trackInitialTarget(event: MouseEvent): void {}
```

Reply:
And move to bottom of the file 😇